packages/x-components/src/components/base-grid.vue

```diff
@@ -158,6 +158,41 @@
         };
       });
     }
+
+    /**
+     * Initialises the rendered columns number and sets a ResizeObserver to keep it updated.
+     *
+     * @internal
+     */
+    protected mounted(): void {
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      const resizeObserver = new ResizeObserver(this.emitRenderedColumnsNumber);
+      resizeObserver.observe(this.$el);
+      this.$on('hook:beforeDestroy', () => {
+        resizeObserver.disconnect();
+      });
+    }
+
+    /**
+     * Emits a {@link XEventsTypes.ColumnsNumberRendered | ColumnsNumberRendered} event.
+     *
+     * @internal
+     */
+    protected emitRenderedColumnsNumber(): void {
+      this.$x.emit('ColumnsNumberRendered', this.getRenderedColumnsNumber());
```

Comment:
Instead of emitting this directly, should we do it through a data property and an `XEmit` decorator? Now every time you resize or load a new page, the same event is emitted lots of times. With this approach, Vue reactivity would avoid emitting the event if it has the same value than the previous one.

![image](https://user-images.githubusercontent.com/68222542/198973330-a3c764c1-0371-4f06-bbb6-e45953ceb193.png)